test/redis_client_mocks/nearby_search_test.go

```diff
@@ -71,23 +71,51 @@ func init() {
 	}
 }
 
-// The setup of this test case guarantees that the Peter Luger's Steakhouse is located OUTSIDE the search radius coverage
+// The setup of this test case guarantees that the Peter Luger's Steakhouse is located OUTSIDE the search radius coverage,
 func TestGetPlaces_shouldExcludePlacesOutsideOfSearchRadius(t *testing.T) {
 	placeSearchRequest := iowrappers.PlaceSearchRequest{
-		Location: POI.Location{Longitude: -74.0060, Latitude: 40.7128},
-		PlaceCat: POI.PlaceCategoryEatery,
-		Radius:   uint(5000),
+		Location:   POI.Location{Longitude: -74.0060, Latitude: 40.7128},
+		PlaceCat:   POI.PlaceCategoryEatery,
+		Radius:     uint(5000),
+		PriceLevel: POI.PriceLevelFour,
 	}
 
 	cachedEateryPlaces, err := RedisClient.NearbySearch(RedisContext, &placeSearchRequest)
 	if err != nil {
 		t.Errorf(err.Error())
 		return
 	}
+	// "Keens Steakhouse"
+	var expect_place = places[2]
 
-	if len(cachedEateryPlaces) != 1 || cachedEateryPlaces[0].ID != places[2].ID {
-		t.Logf("number of nearby eatery places obtained from Redis is %d", len(cachedEateryPlaces))
-		t.Error("failed to get cached Eatery place")
+	if len(cachedEateryPlaces) != 1 {
+		t.Errorf("Expect to have 1 place, but got %d instead", len(cachedEateryPlaces))
+		return
+	}
+	if cachedEateryPlaces[0].ID != expect_place.ID {
+		t.Errorf("Expect to get %s, but got %s instead", expect_place.Name, cachedEateryPlaces[0].Name)
+		return
+	}
+}
+
+func TestGetPlaces_resultShouldBeEmptyAfterPriceMatch(t *testing.T) {
+	// expect result should be empty, because mock data has no PriceLevelTwo places.
+	placeSearchRequest := iowrappers.PlaceSearchRequest{
+		Location:   POI.Location{Longitude: -74.0060, Latitude: 40.7128},
+		PlaceCat:   POI.PlaceCategoryEatery,
+		Radius:     uint(5000),
+		PriceLevel: POI.PriceLevelTwo,
+	}
+
+	cachedEateryPlaces, err := RedisClient.NearbySearch(RedisContext, &placeSearchRequest)
+	if err != nil {
+		t.Errorf(err.Error())
```

Comment:
Use `t.Error(err)` when no string interpolation is needed.

Reply:
Done